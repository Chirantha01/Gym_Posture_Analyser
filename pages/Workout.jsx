import React from 'react';
import { View, Text, Image, StyleSheet, FlatList , StatusBar } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';

const Workout = () => {
  const data = [
    {
      id: '1',
      image: require('../assets/img-bicep.jpg'),
      title: 'Bicep Curls',
      subtitle: 'Arm Exercise'
    },
    {
      id: '2',
      image: require('../assets/img-squat.jpg'),
      title: 'Squats',
      subtitle: 'Leg Exercises'
    },
    {
      id: '3',
      image: require('../assets/img-sideLateral.jpg'),
      title: 'Side Lateral',
      subtitle: 'Shoulder Exercises'
    },
    {
      id: '4',
      image: require('../assets/img-DumbellPress.jpg'),
      title: 'Dumbell Press',
      subtitle: 'Chest Exercises'
    },
    {
      id: '5',
      image: require('../assets/img-plank.jpg'),
      title: 'Plank',
      subtitle: 'Abdominal Exercises'
    },
    {
      id: '6',
      image: require('../assets/img-SumoSquat.jpg'),
      title: 'Sumo Squat',
      subtitle: 'Leg Exercises'
    },
    {
      id: '7',
      image: require('../assets/img-BenchPress.jpg'),
      title: 'Bench Press',
      subtitle: 'Chest Exercises'
    },
    {
      id: '8',
      image: require('../assets/img-BarbelCurl.jpg'),
      title: 'Barbel Curl',
      subtitle: 'Arm Exercises'
    },
  ];


  const renderItem = ({ item, index }) => {
    return (
      <View style={styles.itemContainer}>
        <Image source={item.image } style={styles.image} />
        <View style={styles.textContainer}>
          <Text style={styles.title}>{item.title}</Text>
          <Text style={styles.subtitle}>{item.subtitle}</Text>
        </View>
      </View>
    );
  };

  return (
    <View>
      <StatusBar style="auto" />
      <FlatList
        data={data}
        contentContainerStyle={styles.container}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
        numColumns={2}
        ListHeaderComponent={() => (
          <Text className="text-3xl font-bold p-6 justify-center bg-white">Workouts</Text>
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor:'#fff'
  },
  itemContainer: {
    width: '48%',
    height: 300,
    margin: '1%',
    backgroundColor: '#f9f9f9',
    borderRadius: 8,
    overflow: 'hidden',
    padding: 16,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection:  'column' 
  },
  image: {
    width: '100%',
    height: 200,
    resizeMode: 'cover',
    borderRadius: 8,
  },
  textContainer: {
    padding: 16,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 18,
  },
  subtitle: {
    fontSize: 14,
    color: 'gray',
  },
});

export default Workout;
