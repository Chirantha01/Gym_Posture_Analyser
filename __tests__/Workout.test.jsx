// Workout.test.jsx
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import Workout from '../pages/Workout'; // Adjust the import based on your project structure

// Mock the useNavigation hook
const mockNavigate = jest.fn(); // Create a mock function for navigation

jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
        navigate: mockNavigate, // Return the mock function
    }),
}));

const renderWithNavigation = (component) => {
    return render(
        <NavigationContainer>
            {component}
        </NavigationContainer>
    );
};

test('renders Workout correctly', () => {
    const { getByText, getByPlaceholderText } = renderWithNavigation(<Workout />);

    // Check if the initial texts are rendered
    expect(getByText(/Select your workout,/i)).toBeTruthy();
    expect(getByText(/We will help you correct your posture.../i)).toBeTruthy();

    // Check if the search input is rendered
    const searchInput = getByPlaceholderText(/Search Workouts.../i);
    expect(searchInput).toBeTruthy();
});

test('searches for workouts correctly', () => {
    const { getByPlaceholderText, getByText, queryByText } = renderWithNavigation(<Workout />);

    const searchInput = getByPlaceholderText(/Search Workouts.../i);

    // Check if all workouts are initially displayed
    expect(getByText(/Bicep Curls/i)).toBeTruthy(); // Check if Bicep Curls is rendered
    expect(getByText(/Squats/i)).toBeTruthy(); // Check if Squats is rendered

    // Search for 'Squats'
    fireEvent.changeText(searchInput, 'Squats');

    // Check if the filtered workout appears
    expect(getByText(/Squats/i)).toBeTruthy(); // Squats should still be visible
    expect(queryByText(/Bicep Curls/i)).toBeNull(); // Bicep Curls should not be displayed

    // Clear the search
    fireEvent.changeText(searchInput, '');

    // Check if all workouts are displayed again
    expect(getByText(/Bicep Curls/i)).toBeTruthy(); // Ensure Bicep Curls is visible again
    expect(getByText(/Squats/i)).toBeTruthy(); // Ensure Squats is still visible
});

test('navigates to workout detail on card press', () => {
    const { getByText } = renderWithNavigation(<Workout />);
    
    // Simulate pressing the "Bicep Curls" workout card
    const bicepCard = getByText(/Bicep Curls/i);
    fireEvent.press(bicepCard);
    
    // Check if the navigation function was called
    expect(mockNavigate).toHaveBeenCalledWith('Model_bicep'); // Ensure navigate was called with the correct argument
});

test('matches snapshot', () => {
    const { toJSON } = renderWithNavigation(<Workout />);
    expect(toJSON()).toMatchSnapshot(); // Snapshot testing
});
