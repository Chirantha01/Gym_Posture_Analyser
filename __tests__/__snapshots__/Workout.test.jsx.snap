// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#232323",
      "flex": 1,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search Workouts..."
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 40,
        "margin": 16,
        "paddingHorizontal": 10,
      }
    }
    value=""
  />
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#896CFE",
        "fontSize": 26,
        "fontWeight": "bold",
        "marginBottom": 0,
        "marginLeft": 20,
      }
    }
  >
    Select your workout,
  </Text>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "white",
        "fontSize": 15,
        "fontWeight": "semi-bold",
        "marginBottom": 20,
        "marginLeft": 20,
      }
    }
  >
    We will help you correct your posture...
  </Text>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 16,
      }
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#f9f9f9",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 140,
            "marginBottom": 16,
            "marginHorizontal": 15,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 16,
            "paddingBottom": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "paddingRight": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "500",
              }
            }
          >
            Bicep Curls
          </Text>
          <Text
            style={
              {
                "color": "gray",
                "fontSize": 14,
              }
            }
          >
            Top Arm Exercise
          </Text>
        </View>
        <Image
          source={
            {
              "testUri": "../../../assets/img-bicep.jpg",
            }
          }
          style={
            {
              "borderRadius": 20,
              "height": 140,
              "resizeMode": "cover",
              "width": 175,
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#f9f9f9",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 140,
            "marginBottom": 16,
            "marginHorizontal": 15,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 16,
            "paddingBottom": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "paddingRight": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "500",
              }
            }
          >
            Squats
          </Text>
          <Text
            style={
              {
                "color": "gray",
                "fontSize": 14,
              }
            }
          >
            Top Leg Exercises
          </Text>
        </View>
        <Image
          source={
            {
              "testUri": "../../../assets/img-squat.jpg",
            }
          }
          style={
            {
              "borderRadius": 20,
              "height": 140,
              "resizeMode": "cover",
              "width": 175,
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#f9f9f9",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 140,
            "marginBottom": 16,
            "marginHorizontal": 15,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 16,
            "paddingBottom": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "paddingRight": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "500",
              }
            }
          >
            Push Ups
          </Text>
          <Text
            style={
              {
                "color": "gray",
                "fontSize": 14,
              }
            }
          >
            Top Chest Exercises
          </Text>
        </View>
        <Image
          source={
            {
              "testUri": "../../../assets/pushup.jpg",
            }
          }
          style={
            {
              "borderRadius": 20,
              "height": 140,
              "resizeMode": "cover",
              "width": 175,
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#f9f9f9",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 140,
            "marginBottom": 16,
            "marginHorizontal": 15,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 16,
            "paddingBottom": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "paddingRight": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "500",
              }
            }
          >
            Lat Pull Down
          </Text>
          <Text
            style={
              {
                "color": "gray",
                "fontSize": 14,
              }
            }
          >
            Top Back Exercises
          </Text>
        </View>
        <Image
          source={
            {
              "testUri": "../../../assets/latPullDown.jpg",
            }
          }
          style={
            {
              "borderRadius": 20,
              "height": 140,
              "resizeMode": "cover",
              "width": 175,
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#f9f9f9",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 140,
            "marginBottom": 16,
            "marginHorizontal": 15,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 16,
            "paddingBottom": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "paddingRight": 16,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "500",
              }
            }
          >
            Plank
          </Text>
          <Text
            style={
              {
                "color": "gray",
                "fontSize": 14,
              }
            }
          >
            Top Ab Exercises
          </Text>
        </View>
        <Image
          source={
            {
              "testUri": "../../../assets/img-plank.jpg",
            }
          }
          style={
            {
              "borderRadius": 20,
              "height": 140,
              "resizeMode": "cover",
              "width": 175,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
